lib/pbench/server/api/resources/upload_api.py

```diff
@@ -117,7 +117,9 @@ def put(self, controller):
 
         # Create a tracking dataset object; it'll begin in UPLOADING state
         try:
-            dataset = Dataset(controller=controller, path=tar_full_path, md5=md5sum)
+            dataset = Dataset(
+                owner="pbench", controller=controller, path=tar_full_path, md5=md5sum
+            )
```

Comment:
I think that really just obscures the temporary hack... however there should at least be a clear `FIXME` comment, and I guess I rushed through this door too quickly.

Reply:
My idea was that the changes made to line now 121 would be "permanent" and the "temporary hack" would be positioned elsewhere, where, at the appropriate moment, it would be replaced wholesale.